### DISCUSSION TITLE
The rdp service is not working

### DISCUSSION THREAD

**Discussion by Abygail007** [Ylianst/MeshCentral#5866] [Q&A]
Hello everybody!

I have a little problem. I have a meshcentral installation on a Linux server. All my Windows 10 machines are integrated into it (280). Everything works great! But there is just the "RDP" connection which fails to work.. I tested without our antivirus system but same problems.

Have you had this problem too?

Thanks for the help!

**Comment by si458** [comment]
How have u added the windows devices?
Did u use the meshagent or as local devices?
Have u checked u can rdp from one computer to another first? To check the port 3389 is open and working?

**Reply by Abygail007** [reply]
I use "tactical RMM". With the installation agent it also installs the "meshcentral" agent. I can connect without problem in "normal" mode and if I use the Windows Server remote desktop connection service it works.

**Reply by si458** [reply]
so you need to check you have enabled the windows rdp https://learn.microsoft.com/en-us/windows-server/remote/remote-desktop-services/clients/remote-desktop-allow-access#windows-10-fall-creator-update-1709-or-later
and also you MIGHT need to untick the NLA if you use azure

**Reply by Abygail007** [reply]
On all Windows 10 the option is activated. And we have a local active directory.
I'm stuck there a little. Port 3389 is open in my firewall.

**Reply by si458** [reply]
whats your config.json file like? (hide secret info)

**Reply by Abygail007** [reply]
thank you very much for your help

in "[Show server configuration]"

```
{
  "settings": {
    "cert": "https://mesh**********",
    "WANonly": true,
    "minify": 1,
    "port": 4430,
    "aliasPort": 443,
    "redirPort": 800,
    "allowLoginToken": true,
    "allowFraming": true,
    "agentPing": 35,
    "allowHighQualityDesktop": true,
    "tlsOffload": "127.0.0.1",
    "agentCoreDump": false,
    "compression": true,
    "wsCompression": true,
    "agentWsCompression": true,
    "maxInvalidLogin": { "time": 5, "count": 5, "coolofftime": 30 },
    "postgres": {
      "user": "*********",
      "password": "****************",
      "port": "****",
      "host": "localhost"
    }
  },
  "domains": {
    "": {
      "title": "Tactical RMM",
      "title2": "Tactical RMM",
      "newAccounts": false,
      "certUrl": "https://mesh***********443/",
      "geoLocation": true,
      "cookieIpCheck": false,
      "mstsc": true
    }
  }
}
```

**Reply by si458** [reply]
a few issues with your config.json but shouldnt effect the rdp
`cert` should just be the hostname, not including ports and https://
`minify` should be true or false, best to set it to false incase the is an issue with the min js files
i would disable `compression, wscompression, agentwscompression` incase the is a compression issue
but other than that, all looks ok?
remember to restart meshcentral if u change any of those values

**Reply by Abygail007** [reply]
I made the changes to the server configuration. a reboot and it didn't change anything.

I really don't see where this could come from. On the Linux firewall I logically have nothing to do about it?

**Reply by Abygail007** [reply]
When I try to connect via RDP, I look in the logs and in information I see an IP address 10.72.80.14 but this is not the IP address of my meshcentral server. Is this normal?

![image](https://github.com/Ylianst/MeshCentral/assets/153526600/b583f083-a8fc-43ea-93d4-06df977aab46)

**Reply by Abygail007** [reply]
To test, I reinstalled meshcentral but on a Windows virtual machine rather than Linux.

And it works for the rdp. But I would like it to work for my Linux (I prefer for the security of my network and to avoid having to reinstall and re-configure everything)

**Reply by si458** [reply]
this to me screams a firewall issue again?
whats the firewall on your linux server?
have u enabled outbound ports 3389?

**Reply by Abygail007** [reply]
On Linux, I have not integrated a firewall. But it seems to me that there is one on the base system ?
The difference is also that I don't have this Linux server integrated into the domain. Would that make a difference?

How can port 3389 be allowed in debian?

**Reply by si458** [reply]
In theory I think linux firewall is off by default?

**Reply by Abygail007** [reply]
I looked at "sudo ufw status"
It's off

**Reply by si458** [reply]
im not 100% sure im sorry
as you can use the 'desktop' connect no problems, 
so the RDP connect button should just work too?

**Reply by Abygail007** [reply]
Is there a possibility to export all the configuration, settings, all user, and agent from meshcentral linux to the meshcentral that I installed under Windows?

**Reply by si458** [reply]
yes,

if you use nedb you can just copy the whole `meshcentral-data` folder from one to other,

if you are moving to different databases, you can export your current db and then import to the new db
https://ylianst.github.io/MeshCentral/meshcentral/#database

**Reply by Abygail007** [reply]
After several tests and research, here is some additional information:

• When I run the devtool, we see that a WEB request (webscoket connection is initiated but it remains in "pending" state for several minutes. (status 101)

• I also see an "RDP-CLOSE" type message in the request.

• There are no other requests made.

• Activation of traces on the MESHCENTRAL server:

![image](https://github.com/Ylianst/MeshCentral/assets/153526600/d9db1273-68ef-4ba1-bd78-95af0cfb1f93)


• We see that the WEBSOCKET connection is closed just after opening

• I do not observe any other logs on this subject (neither on the NGINX side, nor from the MESHCENTRAL log files)

• I have the impression that the browser which initiates the WEBSOCKET cannot obtain a response from the server (here the NODEJS RDP server).

• In terms of installed NODEJS modules, I cannot find the NODE-RDPJS module present.

• On the application's documentation and forum, I haven't seen anything about adding and re-registering a nodejs package.
Moreover, the MESHCENTRAL package present on the server is not exactly the same as that present on Github.


• On the client machines (the RDP connections have been tested to the server on which I installed the MESHAGENT), I cannot find any RDP connection attempt. This is why I think there is no RDP connection attempt initiated by the MESHCENTRAL server. There is a problem upstream, as evidenced by the WEBSOCKET non-response problem. For my part, a NODEJS dependency must be missing on the RMM Linux server.

• In order to test an RDP connection from an environment other than MESHCENTRAL, I install an RDP client on the RMM server: freeRDP

• Testing a connection via the freeRDP client: OK, the RDP connection is working correctly.


So ! I don't know... Do you have any ideas?

**Reply by drissm69** [reply]
Hello , i have the same problem , did you find something ?
